### PR TITLE
covered test for injecting spans

### PR DESCRIPTION
**Pull request by YANYZP**
test for https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/63

### CODE REVIEW

**Comment by YANYZP** [comment]
Ready for final review

File: exporter/trace/trace_proto.go [hunk 1]
```diff
@@ -71,22 +71,21 @@ func generateDisplayName(s *export.SpanData, format DisplayNameFormatter) string
 	}
 }
 
-
-// If there are duplicate keys present in the list of attributes, 
-// then the first value found for the key is preserved.
+// If there are duplicate keys present in the list of attributes,
+// then the last value found for the key is preserved.
 func injectLabelsFromResources(sd *export.SpanData) {
```
Comment (YANYZP) [inline exporter/trace/trace_proto.go:76]: Even though the logic is to preserve the first value, the result would be the last value because Resource package would overwrite the values. https://github.com/open-telemetry/opentelemetry-go/blob/v0.7.0/sdk/resource/resource.go#L40

Comment (nilebox) [reply exporter/trace/trace_proto.go:76]: You're not invoking the `New` function here, so I don't understand why did you rewrite the comment. In this function the first value is preserved, no?

File: exporter/trace/trace_proto_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		},
+		{
+			export.SpanData{
+				Resource:   resource.New(kv.String("b", "2")),
+				Attributes: []kv.KeyValue{},
+			},
+			export.SpanData{
+				Resource: resource.New(kv.String("b", "2")),
+				Attributes: []kv.KeyValue{
+					kv.String("b", "2"),
```
Comment (nilebox) [inline exporter/trace/trace_proto_test.go:53]: This value should be different from the one in the Resource, to be able to test which one wins.

File: exporter/trace/trace_proto_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			export.SpanData{
+				Resource: resource.New(),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+				},
+			},
+			export.SpanData{
+				Resource: resource.New(),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
```
Comment (nilebox) [inline exporter/trace/trace_proto_test.go:41]: This is exact copy of the span above. Maybe make values different?

File: exporter/trace/trace_proto_test.go (hunk 2 above)
Comment (nilebox) [reply exporter/trace/trace_proto_test.go:53]: Sorry, just realized that this is "expected".
Please add field names everywhere to make this easier to read / less error prone, i.e.
```
{
  input: ...
  expected: ....
}
```

File: exporter/trace/trace_proto_test.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	"testing"
+
+	export "go.opentelemetry.io/otel/sdk/export/trace"
+	"go.opentelemetry.io/otel/sdk/resource"
+
+	"go.opentelemetry.io/otel/api/kv"
+)
+
+func TestInjectLabelsFromResources(t *testing.T) {
+	testcases := []struct {
```
Comment (nilebox) [inline exporter/trace/trace_proto_test.go:27]: I don't see any test cases with conflict between resource and span attributes, e.g.
```go
			export.SpanData{
				Resource: resource.New(kv.String("a", "resource-value")),
				Attributes: []kv.KeyValue{
					kv.String("a", "attribute-value"),
				},
			},

```

File: exporter/trace/trace_proto_test.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+					kv.String("b", "2"),
+				},
+			},
+		},
+		{
+			export.SpanData{
+				Resource: resource.New(
+					kv.String("c", "1"),
+					kv.String("b", "1"),
+					kv.String("b", "2"),
```
Comment (nilebox) [inline exporter/trace/trace_proto_test.go:77]: there's no need to test duplicate keys in `resource.New`, it should be already tested in `opentelemetry-go`.

Comment (nilebox) [reply exporter/trace/trace_proto_test.go:77]: Instead, you may add a test case where `Attributes` slice below has duplicate keys, since you construct it manually.

File: exporter/trace/trace_proto_test.go (hunk 2 above)
Comment (YANYZP) [reply exporter/trace/trace_proto_test.go:53]: No problem

**Comment by YANYZP** [comment]
Ready for final review

File: exporter/trace/trace_proto_test.go [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+					kv.String("b", "3"),
+					kv.String("c", "1"),
+				},
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.input.Resource.String(), func(t *testing.T) {
```
Comment (james-bebbington) [inline exporter/trace/trace_proto_test.go:117]: It's a bit weird to make the resource string the test case name here. A couple of your tests have the same resource so presumably the same test case name.

It would be better to make your comment for each case the name instead, i.e.

```go
testcases := []struct {
	name     string
	input    export.SpanData
	expected export.SpanData
}{
	{
		name: empty resource,
		input: ...
...
}

...

t.Run(tc.name, ...
```

**Comment by YANYZP** [comment]
Real Final review
